src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -142,6 +146,8 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <_OutputPathWasMissing Condition="'$(_OriginalPlatform)' != '' and '$(_OriginalConfiguration)' != '' and '$(OutputPath)' == ''">true</_OutputPathWasMissing>
     <!-- Check whether BaseOutputPath was specified -->
     <BaseOutputPathWasSpecified Condition="'$(BaseOutputPath)' != ''">true</BaseOutputPathWasSpecified>
+    <!-- Initialize BuildDir when empty, to ensure build doesn't produce inconsistent paths -->
+    <BuildDir>$([MSBuild]::EnsureTrailingSlash($([MSBuild]::ValueOrDefault('$(BuildDir)', 'build'))))</BuildDir>
```

Comment:
If you put `build\` instead of `build`, it'll create one fewer string in the default case.

Also, is that the right default anyway? If we don't want people to have to change anything, and it's currently going to bin, I'd say the default should be empty—that way, it wouldn't change behavior unless the user wants it to.

I glanced at some of the test failures, and the top one was BothBaseOutputPathAndOutputPathWereNotSpecified saying it should be bin, not build\bin.

Reply:
I commented this in a different section and I'm seeing that we need to nail down what `BuildDir` is intended to be. https://github.com/dotnet/msbuild/issues/3497#issuecomment-570268502

Daniel suggested that BaseOutputPath and BaseIntermediateOutputPath should be build off of BuildDir. So it sounds to me like BuildDir should just be (by default) the project directory. BOP and BIOP should just default to BuildDir/bin or BuildDir/obj.

Reply:
@Forgind
> If you put `build\` instead of `build`, it'll create one fewer string in the default case.

True but didn't want to hardcode slashes.

> If we don't want people to have to change anything

This is just one way of doing things, but if we do want to preserve older path spec... we can separate out the older logic and newer logic into a separate props/targets and switch between them with a property say, `UseNewOutputPaths`

@BenVillalobos I thought of that too. My first iteration was the same but this one puts everything under a single folder under project root or solution root.

One of the things we can do with this, is, we can now delete build without publish and vice-versa which many seem to prefer. With this we can also do a force clean which just deletes `build` folder without us having to manually delete `bin`/`obj` folders.